ambry-store/src/test/java/com.github.ambry.store/IndexValueTest.java

```diff
@@ -53,30 +91,50 @@ public void deleteRecordTest() {
     String logSegmentName = LogSegmentNameHelper.getName(pos, gen);
     long oldSize = Utils.getRandomLong(TestUtils.RANDOM, 1000);
     long oldOffset = Utils.getRandomLong(TestUtils.RANDOM, 1000);
-    long expiresAtMs = Utils.getRandomLong(TestUtils.RANDOM, 1000);
-    IndexValue value = new IndexValue(oldSize, new Offset(logSegmentName, oldOffset), expiresAtMs);
-
+    long expiresAtMs = Utils.getRandomLong(TestUtils.RANDOM, 1000000);
+    short serviceId = Utils.getRandomShort(TestUtils.RANDOM);
+    short containerId = Utils.getRandomShort(TestUtils.RANDOM);
+    IndexValue value =
+        IndexValueUtils.getIndexValue(oldSize, new Offset(logSegmentName, oldOffset), expiresAtMs, time.seconds(),
+            serviceId, containerId, version);
     long newOffset = Utils.getRandomLong(TestUtils.RANDOM, 1000);
     long newSize = Utils.getRandomLong(TestUtils.RANDOM, 1000);
-
-    IndexValue newValue = new IndexValue(logSegmentName, value.getBytes());
+    IndexValue newValue = new IndexValue(logSegmentName, value.getBytes(), version);
     // delete in the same log segment
     newValue.setFlag(IndexValue.Flags.Delete_Index);
     newValue.setNewOffset(new Offset(logSegmentName, newOffset));
     newValue.setNewSize(newSize);
-    verifyIndexValue(newValue, logSegmentName, newSize, newOffset, true, expiresAtMs, oldOffset);
+    if (version == 0) {
+      verifyIndexValue(newValue, logSegmentName, newSize, newOffset, true, expiresAtMs, oldOffset, Utils.Infinite_Time,
+          DEFAULT_SHORT_VALUE, DEFAULT_SHORT_VALUE);
+    } else {
+      verifyIndexValue(newValue, logSegmentName, newSize, newOffset, true, expiresAtMs, oldOffset, time.seconds(),
```

Comment:
You can pass the correct value then? Forcing `if` conditions on everyone who needs to use the function seems worse to me.